### ISSUE TITLE
capture queryset with column names

### ISSUE THREAD

**Issue by soheildsh** [trinodb/trino-python-client#101]
I think there should be a way to fetch the query results as an array of dictionaries, which the keys are column names. this has a great use for me and maybe many others. is there any solution for this?
And if there is not would it be ok if I try to contribute something like this to the project?

**Comment by ebyhr** [comment]
Looks good idea and we are welcome your contribution! As you may already know, some DB-API libraries' have such extension. Please keep in mind the backward compatibility when developing the feature.
https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlcursordict.html
https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.row_factory

**Comment by gscteam** [comment]
is this related to this issue #34  ?

**Comment by ebyhr** [comment]
@gscteam No, this is enhancement for query results regardless of the type.

**Comment by scalame** [comment]
@soheildsh how did you manage to get around this limitation for time being?
